### DISCUSSION TITLE
How to create stacked grouped bar chart with visx?

### DISCUSSION THREAD

**Discussion by codesfromshad** [airbnb/visx#1732] [Q&A] [answered]
Hey, guys! I have been trying to figure out a way to create a stacked grouped bar chart with visx. However, since no example is out there, I haven't been able to create one yet.

I was hoping for something like this:
![image](https://github.com/airbnb/visx/assets/92628452/a9ed1ed9-11da-44a7-bea6-e27b4a9b6a33)

But the x-axis will be a dateScale.

**Comment by tkdgus96** [comment, marked as answer]
<img width="936" alt="image" src="https://github.com/airbnb/visx/assets/29229703/0fb56b1f-0f85-4aae-b917-751e6f8eae39">

The first step of doing this is to make the data friendly to this chart. I made the data like this
`[ [first data group], [second data group], [third data group] ]`

The trick is to control the width by the length of your data. For example, if we want to visualize 3 groups of stacked bars, simply divide the width of each bar by 3 and multiply some constant between 0 and 1 (since bars need space from each other). Then, we need to control the X position. Loop the total data and subtract or add a multiple of the bar width to the X position depending on the group's index.

Here is a [sandbox link](https://codesandbox.io/p/sandbox/rough-frog-k2x58n?file=%2FExample.tsx%3A207%2C27&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clryew7fb00063b6kghqt8v8n%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clryew7fb00023b6k3051q3gn%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clryew7fb00033b6kk66ezs50%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clryew7fb00053b6k9ap595r0%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clryew7fb00023b6k3051q3gn%2522%253A%257B%2522id%2522%253A%2522clryew7fb00023b6k3051q3gn%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clryewzpu005c3b6k9auk82me%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FExample.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A207%252C%2522startColumn%2522%253A27%252C%2522endLineNumber%2522%253A207%252C%2522endColumn%2522%253A27%257D%255D%257D%255D%252C%2522activeTabId%2522%253A%2522clryewzpu005c3b6k9auk82me%2522%257D%252C%2522clryew7fb00053b6k9ap595r0%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clryew7fb00043b6k98i14sxc%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clryew7fb00053b6k9ap595r0%2522%252C%2522activeTabId%2522%253A%2522clryew7fb00043b6k98i14sxc%2522%257D%252C%2522clryew7fb00033b6kk66ezs50%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clryew7fb00033b6kk66ezs50%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) demonstrating how you can create a stacked group bar chart with visx.

**Reply by codesfromshad** [reply]
Oh, wow! Thank you so much @tkdgus96 for your help. I am gonna try this out.
